contract/AElf.Contracts.MultiToken/TokenContract_ACS2_StatePathsProvider.cs

```diff
@@ -45,7 +45,7 @@ public override ResourceInfo GetResourceInfo(Transaction txn)
             }
         }
 
-        private int GetPathHashCode(params string[] parts)
+        private ScopedStatePath GetPath(params string[] parts)
```

Comment:
why do we use string path instead of int hash value？

Reply:
We can use string path to get resource keys.However hash value can't.

Reply:
Why do we need to get the resource keys? we just need to know, same resource cannot running in parallel, int value compares other int value much faster than compare string or bytes array.

can you make a performance benchmark compare when you finish a big changes, or make the performance test in the CI? @rosona @jason-hoopox